src/sentry/integrations/discord/actions/issue_alert/form.py

```diff
@@ -45,12 +46,14 @@ def clean(self) -> dict[str, object] | None:
 
         if channel_id and isinstance(channel_id, str):
             try:
+                channel = get_channel_id_from_url(channel_id)
                 validate_channel_id(
-                    channel_id=channel_id,
+                    channel_id=channel,
                     guild_id=integration.external_id,
                     integration_id=integration.id,
                     guild_name=integration.name,
                 )
+                cleaned_data["channel_id"] = channel
```

Comment:
This was missing in previous PR. Since cleaned_data is returned, the channel_id needs to be updated with the cleaned/extracted id.